sdk/core/Azure.Core/src/Response{T}.cs

```diff
@@ -1,12 +1,11 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
-using System;
 using System.ComponentModel;
 
 namespace Azure
 {
-    public readonly struct Response<T>
+    public class Response<T>
```

Comment:
Should we make things `virtual` now that it's a `class`?  Like `Value`?  Or `GetRawResponse()`?

Reply:
Or should it even be `sealed`? I mean, it's going from a `readonly struct` so making members `virtual` - while allowing for flexibility (the point behind this, from the bug) - is going completely the opposite direction. Except for mocking (and a mock could return a `Response<T>`, is there any reason (or even way) a consumer would extend `Response<T>`?

Reply:
For now, the only way to construct `Response<T>` is to pass both. So making virtuals usable we would need to add a protected parameterless ctor. But maybe future-proofing is worth it because making members virtual is a subtle runtime breaking change.

Reply:
Yeah, we already have partner teams (Cosmos) that want to extend Response<T> and add service-specific information there.